src/azure-cli/azure/cli/command_modules/acr/_validators.py

```diff
@@ -85,3 +88,14 @@ def validate_retention_days(namespace):
     days = namespace.days
     if days and (days < 0 or days > 365):
         raise CLIError("Invalid value for days: should be from 0 to 365")
+
+
+def validate_registry_name(cmd, namespace):
+    """Omit login server endpoint suffix."""
+    registry = namespace.registry_name
+    if registry:
+        suffix = cmd.cli_ctx.cloud.suffixes.acr_login_server_endpoint
```

Comment:
seems not all clouds define the `acr_login_server_endpoint`(AzureGermanCloud), it's better to have an empty check here.

Reply:
Thanks!

Reply:
AzureGermanCloud is defined here, https://github.com/Azure/azure-cli/blob/31386c8391217f2a5fcdf0016e80082c81a6ae30/src/azure-cli-core/azure/cli/core/cloud.py#L312. You can also find the other default clouds definition in the same file.

Reply:
It's very helpful, thanks a lot!